types/lhr/audit-details.d.ts

```diff
@@ -79,6 +80,25 @@ declare module Details {
     data: string;
   }
 
+  /**
+   * Entities (1P and 3P) classified and exposed in LHR.
+   */
+  interface EntityClassificationEntity {
+    name: string;
+    company?: string;
+    homepage?: string;
+    isFirstParty?: boolean;
+    isUnrecognized?: boolean;
+  }
+
+  interface EntityClassification {
+    type: 'entity-classification';
+    entities: Array<EntityClassificationEntity>;
+    // The two lookup tables (LUT) below provide O(1) index lookup into entities above.
+    nameLUT: Record<string, number>;
+    originLUT: Record<string, number>;
```

Comment:
Unclear why we'd need this, what's the use case? What's the concern wrt performance? Seems like it'd be negligible. Is this meant to just help HTTP Archive queries?

Reply:
These are used during report rendering — to lookup an entity's name → entity, or origin → entity. [See an example code](https://github.com/GoogleChrome/lighthouse/blob/69509fd3a4b2388a713206d0e191eb3b484c55f0/report/renderer/details-renderer.js#L400) that computes aggregation and maps `item.entity` name to its corresponding heading row. This was initially used much more when I was doing entity recognition only during reporting — looking up URL origins to entities, and entity names to origins, all without increasing JSON size with entity names as keys and avoiding several O(n) loops through entities. 

That said, after reviewing the latest code in the branch now, I'm using it in (only) three places, which could be rewritten:

1. Two instances of `nameLUT`, one during aggregation of table items, and another during rendering of a table row. Both cases avoid looping through all entities each time a row has to be processed.
2. `originLUT` for looking up the first party origin to its corresponding entity. 

Given that its (very) minimal usage, I think it might make sense to make the JSON output of the audit:
```
{
   entities: Record<string, EntityClassificationEntity>`,
   firstParty: string, 
} 
```
and that would solve for the two instances of entity name → entity lookup, still without a perf penalty. The first party name being a string would allow for a direct lookup from `entities` record as well. tl;dr: We could drop both the lookup tables. 

wdyt?

Reply:
Updated PR with the new classification output (and updated the grouping branch, it . I think it looks better, thanks for questioning that!